backends/dpdk/dpdkHelpers.cpp

```diff
@@ -159,10 +159,80 @@ bool ConvertStatementToDpdk::preorder(const IR::AssignmentStatement *a) {
         auto mi = P4::MethodInstance::resolve(m, refmap, typemap);
         if (auto e = mi->to<P4::ExternMethod>()) {
             if (e->originalExternType->getName().name == "Hash") {
+                auto di = e->object->to<IR::Declaration_Instance>();
+                auto declArgs = di->arguments;
+                if (declArgs->size() == 0) {
+                    ::error(ErrorType::ERR_UNEXPECTED, "Expected atleast 1 argument for %1%",
+                                e->object->getName());
+                        return false;
+                }
+                auto hash_alg = declArgs->at(0)->expression;
+                unsigned hashAlgValue = 0;
+                if (hash_alg->is<IR::Constant>())
+                    hashAlgValue = hash_alg->to<IR::Constant>()->asUnsigned();
+                cstring hashAlgName = "";
+                if (hashAlgValue == JHASH0 || hashAlgValue == JHASH5)
+                    hashAlgName = "jhash";
+                else if (hashAlgValue >= CRC1 && hashAlgValue <= CRC4)
+                    hashAlgName = "crc32";
+
+                IR::Vector<IR::Expression> components;
+                IR::ListExpression *listExp = nullptr;
+
+                /* This process the two prototype for Hash Extern
+                    /// Constructor
+                    Hash(PSA_HashAlgorithm_t algo);
+
+                    1. Compute the hash for data.
+                        O get_hash<D>(in D data);
+
+                    2. Compute the hash for data, with modulo by max, then add base.
+                        O get_hash<T, D>(in T base, in D data, in T max);
+
+                        Here, DPDK targets allows constant expression for
```

Comment:
add only to the comment here
DPDK targets allows "only" constant expression for